src/lazyloading.js

```diff
@@ -20,117 +25,127 @@ const observerDefault = {
-              class: this.$el.getAttribute('class'),
-          };
+        const observer = new IntersectionObserver(this.checkIsIntersecting, this.options);
 
-          this.changeElementAttribute(
-              ['class', 'style', 'src'],
-              [
-                  this.placeholderImage.class,
-                  this.placeholderImage.style,
-                  this.placeholderImage.src,
-              ]
-          );
-    },
-    /**
-     * Load image from finish observer
-     * @returns {null}
-     * */
-    loadImage() {
-        this.$el.src = this.image.src;
-        this.firstStepToLoad = false;
-    },
-    /**
-     * Event on load image
-     * @returns {null}
-     * */
-    onLoadImage() {
-      if (!this.firstStepToLoad) {
-          this.changeElementAttribute(['class', 'style'], [this.image.class, '']);
-      }
+        observer.observe(this.$el);
     },
-    /**
-     * Event on error image
-     * @returns {null}
-     * */
-    onErrorImage() {
-        this.changeElementAttribute(
-            ['class', 'style', 'src'],
-            [this.image.class, '', require('./assets/img/photo.png')]
-        );
-    },
-    /**
-     * Method to check intersecting event
-     * @params {Array, Observer} entries, observer
-     * @returns {null}
-     * */
-    checkIsIntersecting(entries, observer) {
-      entries.forEach((entry) => {
-        if (entry.isIntersecting) {
-            this.loadImage();
-            observer.unobserve(this.$el);
-        }
-      })
+
+    methods: {
+        /**
+         * Start data to component
+         * @returns {null}
+         * */
+        setImageData() {
+            this.image = {
+                src: this.$el.getAttribute('src'),
+                class: this.$el.getAttribute('class'),
+            };
+
+            this.changeElementAttribute(
+                ['class', 'style', 'src'],
+                [
+                    this.placeholderImage.class,
+                    this.placeholderImage.style,
+                    this.placeholderImage.src,
+                ],
+            );
+        },
+        /**
+         * Load image from finish observer
+         * @returns {null}
+         * */
+        loadImage() {
+            this.$el.src = this.image.src;
+            this.firstStepToLoad = false;
+        },
+        /**
+         * Event on load image
+         * @returns {null}
+         * */
+        onLoadImage() {
+            if (!this.firstStepToLoad) {
+                this.changeElementAttribute(['class', 'style'], [this.image.class, '']);
+            }
+        },
+
+        // TODO: onError there should be an image from props or default
+        /**
+         * Event on error image
+         * @returns {null}
+         * */
+        onErrorImage() {
+            this.changeElementAttribute(
+                ['class', 'style', 'src'],
+                [this.image.class, '', require('./assets/img/photo.png')],
+            );
+        },
```

Comment:
Can your change require('./assets/img/photo.png') to dynamic?

Reply:
ok!